streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBuffer.java

```diff
@@ -206,6 +291,11 @@ private void logValue(final Bytes key, final BufferKey bufferKey, final BufferVa
         final ByteBuffer buffer = value.serialize(sizeOfBufferTime);
         buffer.putLong(bufferKey.time());
         final byte[] array = buffer.array();
+        this.context = ProcessorContextUtils.asInternalProcessorContext(wrapped().context);
```

Comment:
Okay, so the reason it is like this is because `getStateStore` in the processor only gives the store rather than the buffer. So the init call isn't keeping the changes here. I can move the field updates to the set serde update. Maybe that is better?